### ISSUE TITLE
Issue running sample problem from the book

### ISSUE THREAD

**Issue by meganbryson** [MOEAFramework/MOEAFramework#400]
Hi,
I'm trying to run the sample problem from the start of the book and I'm running into an issue. The error is:

Exception in thread "main" java.lang.Error: Unresolved compilation problems: 
	The type org.apache.commons.lang3.reflect.ConstructorUtils is not accessible
	ConstructorUtils cannot be resolved

	at MOEAFramework/org.moeaframework.ProblemBuilder.<init>(ProblemBuilder.java:24)
	at MOEAFramework/org.moeaframework.Executor.<init>(Executor.java:158)
	at MOEAFramework/chapter2.RunSchafferProblem.main(RunSchafferProblem.java:10)

Lang3 is in the lib and I've even tried replacing it with a new download and downloading Apache to my Eclipse and a few other random things. I'm not sure what I'm doing wrong, do you have any insights?

**Comment by meganbryson** [comment]
Right now I just trying the sample problem. Eventually, I'm trying to recreate the results from [this paper](https://www.sciencedirect.com/science/article/pii/S0377221717307841) for my grad school comb-ops course. They used your framework in the paper :)

**Comment by dhadka** [comment]
I suspect the issue is the Java Classpath isn't updated to reference the new versions of the libraries.  If using Eclipse:

1. Right-click on the project and select Build Path -> Configure Build Path.
2. Remove any old libraries in the list (should have a ❌ next to it) and add the new versions

If that doesn't work, please provide a screenshot of the contents of the `lib/` folder so I can see what versions are being used.

Also FYI, I've been adding newer examples and documentation to https://github.com/MOEAFramework/MOEAFramework/blob/master/docs/README.md.

**Comment by meganbryson** [comment]
<img width="1003" alt="image" src="https://github.com/MOEAFramework/MOEAFramework/assets/72516063/eae737f2-51c8-41e9-b105-526c3c7551ae">
<img width="688" alt="image" src="https://github.com/MOEAFramework/MOEAFramework/assets/72516063/14189d06-dd22-4069-a38a-987d3fac8e97">
I will check out those newer examples too! Thank you :)

**Comment by dhadka** [comment]
You can also completely skip the code that is giving you that error by replacing that example with:

```java
NSGAII algorithm = new NSGAII(new SchafferProblem());
algorithm.run(10000);
algorithm.getResult().display();
```

☝  is the newer and simpler way to setup, run, and view the results.

**Comment by meganbryson** [comment]
The problem tab says the issue is with BasicStroke in ApproximationSetViewer.java
<img width="946" alt="image" src="https://github.com/MOEAFramework/MOEAFramework/assets/72516063/9c13a694-7e48-4084-b441-cd4945ab367b">

**Comment by dhadka** [comment]
That missing `java.awt.BasicStroke` looks to be an issue with Java.

Could you please try pointing Eclipse to a different version of Java?  Go to `Window -> Preferences` and select a different option from this list:

![Image](https://github.com/MOEAFramework/MOEAFramework/assets/2496211/815871af-4540-4935-ae6e-1779fd9c4c72)

If you only have the one Java version installed, I would try installing a different version.  I've been using Azul Zulu - https://www.azul.com/downloads/?package=jdk#zulu.  Just be sure to download the **JDK**.  Use the `Add` button in the dialog above to add the new version to Eclipse.

**Comment by dhadka** [comment]
@meganbryson Any luck fixing this issue?

**Comment by meganbryson** [comment]
Hi David,
Sorry, I didn't see your last comment for some reason. Apologies. 
I switched to Azul Zulu like you recommended (added to my Eclipse, added as a path variable, computer and Eclipse restarted, project cleaned)
<img width="725" alt="image" src="https://github.com/MOEAFramework/MOEAFramework/assets/72516063/8e2fdc13-af83-4957-97c7-623e13c5ab0f">
But now I'm getting 6,000+ errors:
<img width="885" alt="image" src="https://github.com/MOEAFramework/MOEAFramework/assets/72516063/5eb7b606-0045-4b44-a304-b040a56a1ce1">
I'm really not sure how. I really really appreciate you taking the time to help with me this weirdness.

**Comment by dhadka** [comment]
Hopefully some good news...

I believe I was able to re-create the issue you're seeing.  The problem is that the Eclipse project is set up to use **modules**.  That's a newer Java feature didn't exist when I wrote this code, so modules aren't supported.

You can confirm this by looking for a file called `src/module-info.java`.

I'd be curious if you remember what steps you followed to set up the Java project?  That way I can at least update the instructions so others can avoid this situation.

All that being said, to fix the issue, I would recommend:

1. Delete the current project in Eclipse.
2. Download the MOEA Framework source code and extract it to a new folder
3. In Eclipse, use "File -> Import -> Existing Projects into Workspace".  Select the new folder we just created.
4. Try running one of the examples

**Comment by dhadka** [comment]
Here's the steps that should be followed to set up the project in Eclipse - https://github.com/MOEAFramework/MOEAFramework/blob/master/docs/setup.md#setting-up-a-java-project

**Comment by meganbryson** [comment]
1. Thank you so much! It's working you are my savour!!
2. I'm trying to think how I opened it, ugh I wish I remembered, it wasn't something I committed to memory.  Clearly I didn't use Open Projects from File System... but I can't figure out how else I would've done it... or why...  Maybe somehow through New?? I wouldn't worry too much about my error, I do somehow seem to have a special knack for breaking things
